### PR TITLE
SALTO-5024: allow idFields fro support address to be username of email

### PR DESCRIPTION
**Pull request by edenhassid**
allow idFields fro support address to be username of email

---

_Additional context for reviewer_

---
_Release Notes_: 
zendesk:
* It is possible to add the email username as idField for support_address using `&username`.
---
_User Notifications_: 
None

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/support_address.ts [hunk 1]
```diff
@@ -95,9 +95,28 @@ const templateToEmail = (change: Change<InstanceElement>, deployTemplateMapping:
...
+    return
+  }
+  instance.value.username = 'INVALID_USERNAME'
+}
+
 /**
- * OnFetch and in onDeploy this filter turns the email in support_address to a template expression with a reference
+ * 1. OnFetch and in onDeploy this filter turns the email in support_address to a template expression with a reference
  * to the brand's subdomain. only for zendesk emails. In preDeploy the template expressions are turned back to string.
+ * 2. OnFetch, from the email username@subdomain.zendesk.com we will extranct the username to a hidden field
```
Comment (netama) [inline packages/zendesk-adapter/src/filters/support_address.ts:119]: typo `extranct` -> `extract`

**Review by netama — APPROVED** [review]
added some comments, pre-approving in case we don't need another round

File: packages/zendesk-adapter/src/filters/support_address.ts [hunk 2]
```diff
@@ -95,9 +95,28 @@ const templateToEmail = (change: Change<InstanceElement>, deployTemplateMapping:
   }
 }
 
+const extractUsernameFromEmail = (instance: InstanceElement): void => {
```
[the remark this answers is not in the document]
Comment (edenhassid) [reply packages/zendesk-adapter/src/filters/support_address.ts:98]: domain or subdomain?

**Review by shir-reifenberg — APPROVED** [review]
LGTM 😎

**Pull request merged**